k4FWCore/components/UniqueIDGenSvc.cpp

```diff
@@ -49,13 +49,15 @@ size_t UniqueIDGenSvc::getUniqueID(uint32_t evt_num, uint32_t run_num, const std
   }
 
   auto                        hash = std::hash<std::bitset<bits64 + bits32 + bits32 + bitsSizeT>>{}(combined_bits);
-  std::lock_guard<std::mutex> lock(m_mutex);
-  if (m_uniqueIDs.contains(hash)) {
+  bool inserted = false;
+  {
+    std::lock_guard<std::mutex> lock(m_mutex);
+    std::tie(std::ignore, inserted),  = m_uniqueIDs.insert(hash);
```

Comment:
```suggestion
    std::tie(std::ignore, inserted) = m_uniqueIDs.insert(hash);
```
Ah sorry, I messed up the original suggestion. This should work.